### PR TITLE
Update torch.mean() description about dtype restriction.

### LINKED ISSUE #120173: torch.mean() documentation does not mention what type of input it should receive
### 📚 The doc issue

Should the docs also mention that the input tensor should only have float or complex?

If you try and pass anything else you are encountered with:
`mean(): could not infer output dtype. Input dtype must be either a floating point or complex dtype. Got: Long`

Even though the limitation is not mentioned in the docs

### Suggest a potential alternative/fix

The docs should mention what kind of tensor `.mean()` accepts.

### PR DESCRIPTION
**Pull request by rambadi**
Fixes #120173

### CODE REVIEW

**Comment by rambadi** [comment]
Hello @davidberard98 - would you be able to review/approve this change?  Thanks!

File: torch/_torch_docs.py [hunk 1]
```diff
@@ -6902,10 +6902,11 @@ def merge_dicts(*dicts):
     r"""
 mean(input, *, dtype=None) -> Tensor
 
-Returns the mean value of all elements in the :attr:`input` tensor.
+Returns the mean value of all elements in the :attr:`input` tensor. Input dtype must be either a floating point or complex dtype.
```
Comment (soulitzer) [inline torch/_torch_docs.py:6905]:
```suggestion
Returns the mean value of all elements in the :attr:`input` tensor. Input must be floating point or complex.
```

File: torch/_torch_docs.py [hunk 2]
```diff
@@ -6902,10 +6902,11 @@ def merge_dicts(*dicts):
...
 
 Args:
-    {input}
+    input (Tensor):
+      the input tensor, must be either a floating point or complex dtype.
```
Comment (soulitzer) [inline torch/_torch_docs.py:6909]:
```suggestion
      the input tensor, either of floating point or complex dtype
```

**Review by soulitzer — COMMENTED** [review]
Thanks! I made some suggestions on wording.

**Comment by rambadi** [comment]
Sure. I have commited the requested changes. Pls review @soulitzer

**Review by soulitzer — APPROVED** [review]
LGTM

**Pull request closed without merging**